### PR TITLE
Make productionPlace multiple, facetable, and enabled for Advanced Search #9253

### LINKED ISSUE #9253: Feature Request/Idea: Make Production Location repeatable, facetable, and enabled for Advanced Search
**Overview of the Feature Request**
Make the metadata field productionPlace in the Citation Metadata block repeatable, facetable, and enabled for Advanced Search

**What kind of user is the feature intended for?**
API User, Curator, Depositor, Guest, Superuser, Sysadmin

**What inspired the request?**
Data contained in a dataset may have been produced at multiple places. Making the field productionPlace repeatable will make it possible to reflect this fact in the dataset metadata. Making the field facetable and enabled for Advanced Search will allow us to customize Dataverse collections more appropriately.

**What existing behavior do you want changed?**
The field not being repeatable, facetable, or enabled for Advanced Search

**Any brand new behavior do you want to add to Dataverse?**
No. Other, similar fields are already repeatable, facetable, and enabled for Advanced Search e.g., Keyword.

**Any related open or closed issues to this feature request?**
No.

### PR DESCRIPTION
**Pull request by philippconzett**
**What this PR does / why we need it**:
Allows depositors to define multiple instances of the metadata field productionPlace in the Citation Metadata block, to refine searches based on the content in the field using the search facets, as well as using the field in the Advanced Search option.

**Which issue(s) this PR closes**:

Closes #9253

**Special notes for your reviewer**:
No.

**Suggestions on how to test this**:
1. Create a test dataset with multiple instances of the productionPlace metadata field.
2. Filter results using the field in the filter facets.
3. Conducting a search using the field in the Advanced Search option.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
No. There are already other metadata fields that are repeatable, facetable, and enabled for Advanced Search.

**Is there a release notes update needed for this change?**:
Should briefly mention that the field now is repeatable, facetable, and enabled for Advanced Search.

**Additional documentation**:
No.

### CODE REVIEW

**Comment by pdurbin** [comment]
@philippconzett and I update the Solr schema.xml in a2f5d12

We did this in docker-aio. I'm not sure why it says "Production Place" instead of "Production Location". (Maybe some cruft on my machine?) That update was made here:

- #8708

Anyway, Solr was happy to take the data once we made that change (making it allow multiples).

Screen Shot 2023-01-06 at 12 29 43 PM

**Comment by pdurbin** [comment]
There's a test failing according to https://jenkins.dataverse.org/blue/organizations/jenkins/IQSS-Dataverse-Develop-PR/detail/PR-9254/3/tests

Screen Shot 2023-01-09 at 6 32 01 AM

**Comment by mreekie** [comment]
Prio meeting with Stefano.
- Moved from Community Backlog to ordered backlog

**Comment by jggautier** [comment]
The changes in the citation.tsv file look good, and the changes in the schema.xml make sense to me, although I know less about that.

I think the release notes just need to say "Production Location", the field's new name, instead of "Production Place", the old name. @philippconzett I can change that if that's okay, unless you'd like to.

**Comment by pdurbin** [comment]
Heads up that we should also fix the failing test I mentioned above.

**Comment by jggautier** [comment]
Oooo, sorry. I saw that but it had been a while since you posted it so I thought it might've been resolved.

Is the failing test saying that the API endpoint for importing a dataset using DDI Codebook xml doesn't like that the Production Location field can have multiple values now? So that code has to be adjusted?

**Comment by pdurbin** [comment]
An xml file needs to be changed (and probably the surrounding tests?).

Here, I guess? https://github.com/IQSS/dataverse/blob/v5.12.1/doc/sphinx-guides/source/_static/api/ddi_dataset.xml#L37

Via https://github.com/IQSS/dataverse/blob/v5.12.1/src/test/java/edu/harvard/iq/dataverse/api/DataversesIT.java#L504

Hopefully this PR isn't a backward incompatible change. importDatasetDDIViaNativeApi is being called in that test.

**Comment by philippconzett** [comment]
@jggautier Thanks for spotting the naming issue in the release notes. I have replaced "Production Place" with "Production Location".

@pdurbin I earlier mentioned that I wanted the field to be included in Advanced search, but I never got the issue and the PR updated accordingly. If still possible, I could do it later today. If this would mess up your ongoing work with the PR, I could submit a new issue/PR about Advanced Search. Please let me know which option you prefer. Thanks!

**Comment by pdurbin** [comment]
@philippconzett I'm not assigned as a reviewer. Let's see what @scolapasta and @jggautier (who are) think.

**Comment by pdurbin** [comment]
Speaking of tests, please don't forget about the failing tests!

Screen Shot 2023-02-01 at 10 55 48 AM

**Comment by philippconzett** [comment]
Thanks, @scolapasta. I've edited the files 9253-productionPlace.md and citation.tsv accordingly, but left the file schema.xml untouched. Please let me know I the changes make sense and if anything needs to be improved.

**Comment by jggautier** [comment]
I've reviewed the release notes and citation.tsv files and they look good to me.

I'm not sure if the schema.xml file needs to be changed anymore. And I'm probably not the best person to resolve the testing failures, so I'm going to remove myself as a reviewer now. Thanks all!

File: doc/release-notes/9253-productionPlace.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## Metadata field Production Location now repeatable, facetable, and enabled for Advanced Search
+This enhancement allows depositors to define multiple instances of the metadata field Production Location in the Citation Metadata block, users to filter search results using the filter facets, and using the field in the Advanced Search option.
+
+## Major Use Cases and Infrastructure Enhancements
+* Data contained in a dataset may have been produced at multiple places. Making the field Production Location repeatable will make it possible to reflect this fact in the dataset metadata. Making the field facetable and enabled for Advanced Search will allow us to customize Dataverse collections more appropriately. (Issue #9253, PR #9254)
```
Comment (sekmiller) [inline doc/release-notes/9253-productionPlace.md:5]: You should include upgrade instructions such as:

### Additional Upgrade Steps

Update the Citation metadata block:

- `wget https://github.com/IQSS/dataverse/releases/download/v5.13/citation.tsv`
- `curl http://localhost:8080/api/admin/datasetfield/load -X POST --data-binary @citation.tsv -H "Content-type: text/tab-separated-values"`

Comment (sekmiller) [inline doc/release-notes/9253-productionPlace.md:5]: You should also include instructions for updating schema.xml

## Additional Release Steps

1\. Replace Solr schema.xml to allow multiple production locations to be used. See specific instructions below for those installations with custom metadata blocks (1a) and those without (1b).

1a\.

For installations with Custom Metadata Blocks:

-stop solr instance (usually service solr stop, depending on solr installation/OS, see the [Installation Guide](https://guides.dataverse.org/en/5.13/installation/prerequisites.html#solr-init-script)

- add the following line to your schema.xml:

    `<field name="productionPlace" type="string" stored="true" indexed="true" multiValued="true"/>`

- restart solr instance (usually service solr start, depending on solr/OS)

1b\. 

For installations without Custom Metadata Blocks:

-stop solr instance (usually service solr stop, depending on solr installation/OS, see the [Installation Guide](https://guides.dataverse.org/en/5.13/installation/prerequisites.html#solr-init-script)

-replace schema.xml

cp /tmp/dvinstall/schema.xml /usr/local/solr/solr-8.11.1/server/solr/collection1/conf

-start solr instance (usually service solr start, depending on solr/OS)

**Comment by sekmiller** [comment]
@philippconzett I will update the ImportDDIServiceBean so that it will know to expect multiple Production locations are possible. I'll also add the upgrade instructions that I mentioned above.

**Comment by philippconzett** [comment]
Thanks for adding this, @sekmiller. (I don't how enough about schema.xml to add anything substantial to this.)

**Review by scolapasta — CHANGES_REQUESTED** [review]
@sekmiller Looks good - can you make one more change and add a second Prod location to the ddi that is used for testing? That way we actually exercise this change.

**Review by scolapasta — APPROVED** [review]

File: doc/release-notes/9253-productionPlace.md (hunk 1 above)
Comment (pdurbin) [reply doc/release-notes/9253-productionPlace.md:5]: @sekmiller took care of this in 5b8e81e. Resolving.

**Pull request merged**